plugins/inputs/logfile/tailersrc_test.go

```diff
@@ -321,3 +542,76 @@ func logLine(s string, l int, t time.Time) string {
 	line = line[:l]
 	return line
 }
+
+func logSpecificLine(s string, t time.Time) string {
+	line := ""
+	if !t.IsZero() {
+		line += t.Format(time.RFC3339) + " "
+	}
+	line += s
+
+	return line
+}
+
+func setupTailer(t *testing.T, filters []*LogFilter, multiLineFn func(string) bool, file, statefile *os.File, done chan struct{}, consumed *int32) {
+	tailer, err := tail.TailFile(file.Name(),
+		tail.Config{
+			ReOpen:      false,
+			Follow:      true,
+			Location:    &tail.SeekInfo{Whence: io.SeekStart, Offset: 0},
+			MustExist:   true,
+			Pipe:        false,
+			Poll:        true,
+			MaxLineSize: defaultMaxEventSize,
+			IsUTF16:     false,
+		})
+
+	if err != nil {
+		t.Errorf("Failed to create tailer src for file %v with error: %v", file, err)
+	}
+
+	config := &FileConfig{
+		LogGroupName:  t.Name(),
+		LogStreamName: t.Name(),
+		Filters: filters,
+	}
+	err = config.init()
+	assert.NoError(t, err)
+	ts := NewTailerSrc(
+		t.Name(),
+		t.Name(),
+		"destination",
+		statefile.Name(),
+		tailer,
+		false, // AutoRemoval
+		multiLineFn,
+		config.Filters,
+		parseRFC3339Timestamp,
+		nil, // encoding
+		defaultMaxEventSize,
+		defaultTruncateSuffix,
+		1,
+	)
+
+	ts.SetOutput(func(evt logs.LogEvent) {
+		if evt == nil {
+			close(done)
+			return
+		}
+		atomic.AddInt32(consumed, 1)
+		evt.Done()
+		// extra sanity check
+		assert.True(t, ShouldPublish(t.Name(), t.Name(), config.Filters, evt))
+	})
+}
+
+func assertExpectedLogsPublished(t *testing.T, total, numConsumed, expectedDropped int) {
+	assert.Equal(t, total - expectedDropped, numConsumed)
+	stats := profiler.Profiler.GetStats()
+	statKey := fmt.Sprintf("logfile_%s_%s_messages_dropped", t.Name(), t.Name())
```

Comment:
Took out an assertion here about the size of the stats map because that just causes rigidity for no real value.